### PR TITLE
fix(select): fixed ion-select should pass through ion-text-wrap class…

### PR DESCRIPTION
**Pull request by telishreyas10**
… to underlying .select-text element (#19949)

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
selected options are truncated with ellipsis - no option to enforce selected options to be wrapped (i.e. for multiple selection)

Issue URL: [https://github.com/telishreyas10/ionic-framework/tree/19949](https://github.com/telishreyas10/ionic-framework/tree/19949)

image

## What is the new behavior?

selected options are no longer truncated with ellipsis  for multiple selection

image

Added following line of code in scss file to fix the issue : 

`:host(.ion-text-wrap) .select-text {
  white-space: normal;
}`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by brandyscarney** [comment]
Thank you for the PR! I made a small change to inherit the `white-space` instead. This would allow `white-space` to be set to any value and still be used. I will bring this up for review with the team. 🙂

File: core/src/components/select/select.scss [hunk 1]
```diff
@@ -77,7 +79,7 @@ button {
 
   text-overflow: ellipsis;
 
-  white-space: nowrap;
+  white-space: inherit;
```
Comment (liamdebeasi) [inline core/src/components/select/select.scss:82]: With Ionic 7 we updated `ion-select` to contain both the native control and the label. Will this impact the label when we sync with the v7 branch?

Comment (brandyscarney) [reply core/src/components/select/select.scss:82]: I tested it out on the v7 feature branch and it seems applying it in the same places does still wrap the label properly.

Comment (liamdebeasi) [reply core/src/components/select/select.scss:82]: One more question -- did you test this with the new select syntax?

```
<ion-select label="My Label">...</ion-select>
```

v7 is backwards compatible with the old select syntax, so you'll need to check using the new syntax too.

Comment (liamdebeasi) [reply core/src/components/select/select.scss:82]: Perfect! Then this change looks good to me 👍

**Review by liamdebeasi — APPROVED** [review]

**Comment by brandyscarney** [comment]
I had to recreate this PR because some of our tests don't work on contributions, but I will make sure to add you as a co-author when we merge it! Thank you! https://github.com/ionic-team/ionic-framework/pull/26973

**Pull request closed without merging**